src/js/components/stock-movement-wizard/outbound/AddItemsPage.jsx

```diff
@@ -905,6 +900,13 @@ class AddItemsPage extends Component {
             // newly saved item to replace its equivalent in state
             const itemToChange = _.last(_.differenceBy(lineItemsBackendData, itemCandidatesToSave, 'id'));
             const lineItemsAfterSave = this.state.values.lineItems.map((item) => {
+              if (
+                parseInt(item.quantityRequested, 10) === 0 &&
+                item.rowSaveStatus === RowSaveStatus.SAVING &&
+                !_.includes(savedItemsIds, item.id)
+              ) {
+                return { ..._.omit(item, ['id', 'statusCode']), disabled: true, rowSaveStatus: RowSaveStatus.SAVED };
+              }
```

Comment:
what does it do? we get rid of `id` and disable the line, mark line as `SAVED`, but what was meant behind the scenes about this?

Reply:
It means that the line has a quantity of 0 (I discussed it with Manon, and she said it should be marked as saved). So, after receiving a response with information that my line was deleted, I am removing `id` and `statusCode` properties. These two properties within the `lineItem` indicate that the line is saved, so I have to delete them, because changing the quantity of previously deleted line should treat this line as a new one instead of edited.

Reply:
but if the item is deleted, why should it be marked as saved and why should the line be **disabled** instead of deleted from the table form? I mean, I understand the logic with the omit, but I don't understand the `disabled` and SAVED in this case, if the item is DELETED, not saved.
Does it mean, even though the backend has already deleted the item, you have to delete it from table manually additionally?

Reply:
All of the items with `qty = 0` will be deleted after going to next step or just saving this step manually. Users can change the quantity to 0 by accident, so why we should delete this line from the table? 

Screenshot from the ticket:
![image](https://github.com/openboxes/openboxes/assets/83239466/0f624906-f79f-4cb1-9e22-3e29505e655e)

Reply:
ok, if so, then I understand now